.github/workflows/main.yml

```diff
@@ -83,3 +83,7 @@ jobs:
     - name: isort
       run: isort --check-only .
       if: always()
+
+    - name: pre-commit hook check
```

Comment:
maybe, using the pre commit for lint we can remove the step for flake8, mypy and isort.

I had some problems in the past with pydocstyle as pre-commit hook, but I can create a follow up PR for that to check if it is working now.